### PR TITLE
approvaltests.cpp: Replace use of ${CONAN_LIBS} in test_package with targets

### PR DESCRIPTION
**Pull request by claremacrae**
Based on the feedback in https://github.com/conan-io/conan-center-index/pull/4366#discussion_r565338135 I've adjusted the approvaltests.cpp test_package to explicitly state which targets are needed in the test_package.

Specify library name and version:  **approvaltests.cpp/all-versions**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by claremacrae** [comment]
I've just realised that this changes the same recipe as #4581 - but there are no changed files in common, so it shouldn't cause any conflicts.

**Review by prince-chrismc — DISMISSED** [review]

**Review by SSE4 — DISMISSED** [review]

File: recipes/approvaltests.cpp/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -8,23 +8,25 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 CONAN_BASIC_SETUP()
 
+find_package(ApprovalTests REQUIRED CONFIG)
+
 if(WITH_BOOSTTEST)
     ADD_EXECUTABLE(test_package_boosttest test_boosttest.cpp)
-    target_link_libraries(test_package_boosttest ${CONAN_LIBS})
+    target_link_libraries(test_package_boosttest ApprovalTests::ApprovalTests Boost::Boost)
```
Comment (ericLemanissier) [inline recipes/approvaltests.cpp/all/test_package/CMakeLists.txt:15]: Is it mandatory to link explicitly with `Boost::Boost` ? isn't boost automatically part of the link as being a requirement of `approvaltests.cpp` package ?
dito for the 4 next changes

Comment (claremacrae) [reply recipes/approvaltests.cpp/all/test_package/CMakeLists.txt]: Thanks. When I made the changes and tested them on my machine, I could have sworn that I got missing symbols for all the 3rd-part libraries except in Boost...

However, I just repeated that check, and *didn't* get the missing symbols - and so I've just pushed a new change, with only the `ApprovalTests::ApprovalTests` stated...

**Review by ericLemanissier — APPROVED** [review]

**Comment by claremacrae** [comment]
> Clare Macrae requested a new review from @SSE4 1 minute ago...

sorry about that - I was wondering what that button did, on my iPad, and now I know...

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by claremacrae** [comment]
@prince-chrismc Hi Chris, GitHub is telling me you left review comments, but I can't see them anywhere... What were they, please?

image

**Comment by prince-chrismc** [comment]
the GitHub tricks!

when you pushed changes... GitHub did this

image

Which means I previously approved but it's no longer valid

image

It how the GitHub API works.

https://api.github.com/repos/conan-io/conan-center-index/pulls/4600

https://api.github.com/repos/conan-io/conan-center-index/pulls/4600/reviews

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**